### ISSUE TITLE
bin/admin/build-news generates empty sections when a project has had no changes.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4191]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#4191|
|Type|defect|
|Created|2009-12-25 19:13:45Z|
|Branch|https://github.com/twisted/twisted/tree/no-news-4191|

We should probably write something like "No significant changes have been made for this release." in this case.

<details><summary>Searchable metadata</summary>

```
trac-id__4191 4191
type__defect defect
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_no_news_4191 branches/no-news-4191
branch_author__jml jml
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1261768425000000 1261768425000000
changetime__1275843912000000 1275843912000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

(In [29216]) Branching to 'no-news-4191'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

This bug was fixed along with bug [#4291](https://github.com/twisted/twisted/issues/4291) as part of [28402]. However, the fix there simply does not generate a news section for projects that have had no changes. A section that says "No significant changes made for this release" would actually be more helpful and less confusing to users, I think.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

> A section that says "No significant changes made for this release" would actually be more helpful and less confusing to users, I think.

+1

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

If there are only "misc" changes in the project's topfiles section, the NEWS entry looks like:

```
Twisted News 10.0.0 (2010-06-06)
================================

Other
-----
 - #7777
```

This should include text like "No significant changes have been made for this release" before listing the miscellaneous changes.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

(In [29217]) If there were no changes, say so. refs [#4191](https://github.com/twisted/twisted/issues/4191)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml removed owner<br>@jml set status to `new`|
|-|-|

I've changed the test that said "omit the section if there are no changes". I think the branch is ready for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to @jml|
|-|-|

+1!

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_9"></a>|@jml set status to `closed`|
|-|-|

(In [29221]) Merge no-news-4191.

 * Author: jml
 * Reviewer: therve
 * Fixes [#4191](https://github.com/twisted/twisted/issues/4191)

When generating NEWS sections for projects with no changes, include a
message saying that the project has had no changes.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
